src/rootfs.rs

```diff
@@ -29,227 +28,461 @@ struct MountOptions<'a> {
     cgroup_ns: bool,
 }
 
-pub fn prepare_rootfs(
-    spec: &Spec,
-    rootfs: &Path,
-    bind_devices: bool,
-    cgroup_ns: bool,
-) -> Result<()> {
-    log::debug!("Prepare rootfs: {:?}", rootfs);
-    let mut flags = MsFlags::MS_REC;
-    let linux = spec.linux().as_ref().context("no linux in spec")?;
-
-    match linux.rootfs_propagation().as_deref() {
-        Some("shared") => flags |= MsFlags::MS_SHARED,
-        Some("private") => flags |= MsFlags::MS_PRIVATE,
-        Some("slave" | "unbindable") | None => flags |= MsFlags::MS_SLAVE,
-        Some(uknown) => bail!("unknown rootfs_propagation: {}", uknown),
+/// Holds information about rootfs
+pub struct RootFS {
+    command: Box<dyn Syscall>,
```

Comment:
Sorry, this s mixed up by my mistake, but let's unify the name to `syscall`.

Reply:
Do you mean the field name from command to syscall?
or something else, please guide.

Reply:
Let's simply change the name `command` to `syscall`.